pipeline/transfer_validator.py

```diff
@@ -33,220 +33,270 @@ def transfer_validator(args):
-                    lb-offset, rescaling*cls_mean_dict[val_type,
-                                                       "unfiltered",
-                                                       spec],
-                    rescaling*cls_std_dict[val_type, "unfiltered", spec],
-                    color="navy", marker=".", markerfacecolor="white",
-                    label=r"Unfiltered decoupled $C_\ell$", ls="None"
-                )
-                main.errorbar(
-                    lb+offset, rescaling*cls_mean_dict[val_type,
-                                                       "filtered",
-                                                       spec],
-                    rescaling*cls_std_dict[val_type, "filtered", spec],
-                    color="darkorange", marker=".", markerfacecolor="white",
-                    label=r"Filtered decoupled $C_\ell$", ls="None"
-                )
-
-                if f1 == f2:
-                    main.set_yscale("log")
-
-                # Plot residuals
-                residual_unfiltered = \
-                    (cls_mean_dict[val_type, "unfiltered", spec] -
-                     cls_theory_binned[val_type, spec]) / \
-                    cls_std_dict[val_type, "unfiltered", spec]
-                residual_filtered = \
-                    (cls_mean_dict[val_type, "filtered", spec] -
-                     cls_theory_binned[val_type, spec]) / \
-                    cls_std_dict[val_type, "filtered", spec]
-
-                sub.axhspan(-2, 2, color="gray", alpha=0.2)
-                sub.axhspan(-1, 1, color="gray", alpha=0.7)
-
-                sub.axhline(0, color="k")
-                sub.plot(lb-offset,
-                         residual_unfiltered * np.sqrt(meta.tf_est_num_sims),
-                         color="navy", marker=".", markerfacecolor="white",
-                         ls="None")
-                sub.plot(lb+offset,
-                         residual_filtered * np.sqrt(meta.tf_est_num_sims),
-                         color="darkorange", marker=".",
-                         markerfacecolor="white", ls="None")
-
-                # Multipole range
-                main.set_xlim(2, meta.lmax)
-                sub.set_xlim(*main.get_xlim())
-
-                # Suplot y range
-                sub.set_ylim((-5., 5.))
-
-                # Cosmetix
-                main.set_title(f1+f2, fontsize=14)
-                if spec == "TT":
-                    main.legend(fontsize=13)
-                main.set_xticklabels([])
-                if id1 != 2:
-                    sub.set_xticklabels([])
-                else:
-                    sub.set_xlabel(r"$\ell$", fontsize=13)
-
-                if id2 == 0:
-                    if isinstance(rescaling, float):
-                        main.set_ylabel(r"$C_\ell$", fontsize=13)
+        for ftag1, ftag2 in filtering_pairs:
+            plt.figure(figsize=(16, 16))
+            grid = plt.GridSpec(9, 3, hspace=0.3, wspace=0.3)
```

Comment:
Similar comment to the one above, should we use [this utility](https://github.com/simonsobs/SOOPERCOOL/blob/887a49a56d38014d89a189c207d6931da3465623/soopercool/utils.py#L479) instead?

Reply:
Yes, the plan is to externalize plotting utilities that are used throughout the pipeline (i.e. ps plot with residual)